browser/ftx/ftx_service_factory.cc

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ftx/ftx_service_factory.h"
+
+#include "brave/browser/profiles/profile_util.h"
+#include "brave/components/ftx/browser/ftx_service.h"
+#include "chrome/browser/profiles/incognito_helpers.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+
+// static
+FTXServiceFactory* FTXServiceFactory::GetInstance() {
+  return base::Singleton<FTXServiceFactory>::get();
+}
+
+// static
+FTXService* FTXServiceFactory::GetForProfile(Profile* profile) {
+  if (!brave::IsRegularProfile(profile)) {
+    return nullptr;
+  }
+
+  return static_cast<FTXService*>(
+      GetInstance()->GetServiceForBrowserContext(profile, true));
+}
+
+FTXServiceFactory::FTXServiceFactory()
+    : BrowserContextKeyedServiceFactory(
+          "FTXService",
+          BrowserContextDependencyManager::GetInstance()) {}
+
+FTXServiceFactory::~FTXServiceFactory() {}
+
+KeyedService* FTXServiceFactory::BuildServiceInstanceFor(
+    content::BrowserContext* context) const {
+  return new FTXService(Profile::FromBrowserContext(context));
```

Comment:
FTXService takes BrowserContext so no reason for `Profile::FromBrowserContext`

Reply:
fixed via https://github.com/brave/brave-core/compare/6f4c503619e82a440ef7db9782288e922c37d5e4..f4d887c0baf39543331107f2e22f9b0ad4986c02